modules/apps/site-navigation/site-navigation-menu-item-display-page/src/main/java/com/liferay/site/navigation/menu/item/display/page/internal/type/DisplayPageTypeSiteNavigationMenuItemType.java

```diff
@@ -195,6 +200,32 @@ public String getLabel(Locale locale) {
 		return _displayPageTypeContext.getLabel(locale);
 	}
 
+	@Override
+	public PersistedModel getModel(
+			UnicodeProperties typeSettingsUnicodeProperties)
+		throws PortalException {
+
+		LayoutDisplayPageObjectProvider<?> layoutDisplayPageObjectProvider =
+			_displayPageTypeContext.getLayoutDisplayPageObjectProvider(
+				GetterUtil.getLong(
+					typeSettingsUnicodeProperties.get("classPK")));
+
+		String className = typeSettingsUnicodeProperties.get("className");
+
+		if (className.equals(FileEntry.class.getName())) {
```

Comment:
I tried simplifying the logic for file entries, see https://github.com/jonathanmccann/liferay-portal/commit/36176b4738c09414a03230011feadde115defa1e